src/plugins/ui_stateful/clipboard.ts

```diff
@@ -43,8 +44,9 @@ export class ClipboardPlugin extends UIPlugin {
   private status: "visible" | "invisible" = "invisible";
   private state?: ClipboardState;
   private lastPasteState?: ClipboardState;
-
+  private allowPaste: boolean = true;
```

Comment:
After discussing this issue of infinite with the team and PO, we think it would be best (and overall simpler) to empty the OS clipboard as long as the standard clipboard after the **paste** of a **cut** command.  While it is not what google sheet does, i does not make much sense to keep pasting the content of a cut - an action that only makes sense inside the spreadsheet itself.

Reply:
Hey, I did actually try emptying the clipboard earlier by making its state undefined if operation is 'cut'. But in that case, call would go to paste from os clipboard and there it would be difficult to distinguish whether the content in clipboard is from outside of the sheet or it is from spreadsheet itself.
I've tried multiple approaches to solve this issue but nothing seems to be effective so far. Do you have any suggestions on how this can be achieved?

Reply:
The idea would be to also clear the **OS** Clipboard after a paste from cut.

You cannot do that from the plugin tho, you have to do that from the grid/menu item. Use `env.clipboard.write` to write stuff from the OS clipboard.

Reply:
@khpa-odoo I don't think you have addressed this point. the idea would be to completely get rid of the 'allowPaste' boolean that you added and just assume that we empty the clipboard (both the one of the plugin and of the os once we successfully paste after a cut.  In order to do that,you should edit the `src/actions/menu_items_actions.ts  function paste()` to clear the OS clipboard one the paste action was successful in the plugin, but only if we come from a CUT operation, for this you have the getter `isCutOperation`